modules/standard/Types.chpl

```diff
@@ -411,6 +411,18 @@ proc max(type t) where isComplexType(t) {
   return x;
 }
 
+proc param_max(type t) param where t == int(8)
```

Comment:
Is there a cleaner way to do this? This seems like a pretty ugly/incomplete way to do this. 

I think I'd prefer that the maximum checks either just be left to occur at runtime (it's not a big cost, and it will get turned off with --fast anyways) or to have this stamped out for all types (and leave it's name as max instead of param_max._